src/LEGO.AsyncAPI/Services/AsyncApiReferenceResolver.cs

```diff
@@ -201,8 +201,17 @@ private void ResolveList<T>(IList<T> list) where T : class, IAsyncApiReferenceab
             }
         }
 
-        private T ResolveReference<T>(AsyncApiReference reference) where T : class, IAsyncApiReferenceable
+        private T ResolveReference<T>(AsyncApiReference reference) where T : class, IAsyncApiReferenceable, new()
         {
+            if (reference.IsExternal)
+            {
+                return new ()
```

Comment:
So my understanding based on the [comments](https://github.com/LEGO/AsyncAPI.NET/pull/129#issuecomment-1759343261) on #129 is that *this* is where we would inject the code to read an external file or remote resource using the proposed `ReferenceLoader`?

Reply:
So basically take the reference loader in the ctor, and use that if isExternal. Yeah